### PR TITLE
VACMS-3775: PHP 7.2 updates

### PR DESCRIPTION
**Pull request by indytechcook**
## Description

See #3775 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [x] Manual Code Review Approved.

### Select Team for PR review

- [x] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [x] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change. 
  - [x] Merge & carry on unburdened by announcements

---
This change is [Reviewable](https://reviewable.io/reviews/department-of-veterans-affairs/va.gov-cms/3946)

### CODE REVIEW

**Comment by indytechcook** [comment]
Related devops PR: https://github.com/department-of-veterans-affairs/devops/pull/8314

**Comment by acrollet** [comment]
Confirmed that the spec'd version is present on the dev environment: 
Screen Shot 2021-01-05 at 1 39 32 PM

**Comment by acrollet** [comment]
QA done on tugboat preview:

- confirmed that user sections block works correctly
- checked node view + revisions sidebar
- checked a front-end preview link
- tested editing a page
- reviewed status report for errors

File: .tugboat/config.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ services:
         # Set unlimited memory limit for PHP CLI to allow Composer to run.
         - echo "memory_limit = -1" >> /usr/local/etc/php/conf.d/php-cli.ini
         - echo "max_execution_time = 1200" >> /usr/local/etc/php/conf.d/my-php.ini
+        # Set session timeout
+        - echo "session.cookie_lifetime = 0" >> /usr/local/etc/php/conf.d/my-php.ini
```
Comment (acrollet) [inline .tugboat/config.yml:42]: I assume this is mirrored in the geerlingguy configs pulled in for BRD? I couldn't find an explicit setting to match.

**Review by acrollet — DISMISSED** [review]

File: .tugboat/config.yml (hunk 1 above)
Comment (indytechcook) [reply .tugboat/config.yml:42]: This moves the config from an `ini_set` in the tugboat.settings.php to the php.ini.  Have a session variable set in the ini_set set off a warning on PHP 7.2

**Comment by indytechcook** [comment]
While it looks like everything passed tests we are getting the following item in logs:

```
Uncaught PHP Exception PHPUnit\Framework\Error\Warning: "session_id(): Cannot change session id when headers already  ┃ sent" at /var/lib/tugboat/docroot/vendor/symfony/http-foundation/Session/Storage/Proxy/AbstractProxy.php line 94 
```

My research leads me to belvie the issue is with phpunit 7 and https://phpunit.de/supported-versions.html

File: .tugboat/config.yml [hunk 2]
```diff
@@ -42,6 +42,8 @@ services:
         # Set unlimited memory limit for PHP CLI to allow Composer to run.
         - echo "memory_limit = -1" >> /usr/local/etc/php/conf.d/php-cli.ini
         - echo "max_execution_time = 1800" >> /usr/local/etc/php/conf.d/my-php.ini
+        # Set session timeout
+        - echo "session.cookie_lifetime = 0" >> /usr/local/etc/php/conf.d/my-php.ini
```
Comment (ElijahLynn) [inline .tugboat/config.yml:46]: Can you copy the comment or the good part from the settings.tugboat.php for the "why" too?

Comment (indytechcook) [inline .tugboat/config.yml:46]:
```suggestion
```

**Comment by ElijahLynn** [comment]
I built and deployed this PR with the DevOps changes to BRD in https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3775 and tests are passing now!

image

**Review by ElijahLynn — APPROVED** [review]
Looks good!

**Comment by ElijahLynn** [comment]
I don't think this is related but worth mentioning:
> Notice: Undefined index: #type in Drupal\Core\Form\FormHelper::processStates() (line 208 of /var/www/cms/docroot/core/lib/Drupal/Core/Form/FormHelper.php)

image

**Pull request merged**